src/dft/descriptor_config_helper.hpp

```diff
@@ -94,18 +94,9 @@ PARAM_TYPE_HELPER(config_param::COMMIT_STATUS, config_value)
 template <config_param Param, precision prec, domain dom>
 void set_value(dft_values<prec, dom>& vals,
                param_type_helper_t<real_helper_t<prec>, Param>&& set_val) {
+    vals.changed.insert(Param);
     if constexpr (Param == config_param::LENGTHS) {
-        int rank = vals.rank;
-        if (set_val == nullptr) {
-            throw mkl::invalid_argument("DFT", "set_value", "Given nullptr.");
-        }
-        for (int i{ 0 }; i < rank; ++i) {
-            if (set_val[i] <= 0) {
-                throw mkl::invalid_argument("DFT", "set_value",
-                                            "Invalid length value (negative or 0).");
-            }
-        }
-        std::copy(set_val, set_val + rank, vals.dimensions.begin());
+        throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
```

Comment:
From our discussions we agreed that length does not need to be read-only for now. We will change the spec if we decide it is needed.

Reply:
This has been reverted